File: LHDS.AdminPortal.Api.Tests.Acceptance/Models/IngestionTrackings/IngestionTracking.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace LHDS.AdminPortal.Api.Tests.Acceptance.Models.IngestionTrackings
{
    public class IngestionTracking
    {
        public Guid Id { get; set; }
        public string FileName { get; set; } = string.Empty;
        public Guid SupplierId { get; set; }
        public string EncryptedFileName { get; set; } = string.Empty;
        public string DecryptedFileName { get; set; } = string.Empty;
        public bool Decrypted { get; set; }

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTimeOffset LastSeen { get; set; }
        public Guid? SubscriberAgreementId { get; set; }

        public bool FileDeleted { get; set; }
        public int EncryptedFileSize { get; set; }
        public int DecryptedFileSize { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public string UpdatedBy { get; set; } = string.Empty;

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTimeOffset UpdatedDate { get; set; }

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTimeOffset CreatedDate { get; set; }
    }
}

File: LHDS.AdminPortal.Web/ClientApp/src/models/ingestionTrackings/ingestionTracking.ts
import { Supplier } from '../suppliers/supplier';

export class IngestionTracking {
    public id: string;
    public fileName: string;
    public supplierId: string;
    public encryptedFileName: string;
    public decryptedFileName: string;
    public decrypted: boolean;
    public lastSeen: Date;
    public fileDeleted: boolean;
    public encryptedFileSize: number;
    public decryptedFileSize: number;
    public isDownloaded: boolean;
    public isProcessing: boolean;
    public retryCount: number;
    public sourceFolderPath: string;
    public lastAttemptedDate: Date;
    public dataSetSpecificationId: string;
    public batch: string;
    public objectName: string;
    public batchReadyFolderPath: string;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;
    public supplier?: Supplier;

    constructor(ingestionTracking: any) {
        this.id = ingestionTracking.id;
        this.fileName = ingestionTracking.fileName || "";
        this.supplierId = ingestionTracking.supplierId || "";
        this.encryptedFileName = ingestionTracking.encryptedFileName || "";
        this.decryptedFileName = ingestionTracking.decryptedFileName || "";
        this.decrypted = ingestionTracking.decrypted;
        this.lastSeen = ingestionTracking.lastSeen ? new Date(ingestionTracking.lastSeen) : new Date();
        this.fileDeleted = ingestionTracking.fileDeleted;
        this.encryptedFileSize = ingestionTracking.encryptedFileSize;
        this.decryptedFileSize = ingestionTracking.decryptedFileSize;
        this.isDownloaded = ingestionTracking.isDownloaded;
        this.isProcessing = ingestionTracking.isProcessing;
        this.retryCount = ingestionTracking.retryCount;
        this.sourceFolderPath = ingestionTracking.sourceFolderPath || "";
        this.lastAttemptedDate = ingestionTracking.lastAttemptedDate ? new Date(ingestionTracking.lastAttemptedDate) : new Date();
        this.dataSetSpecificationId = ingestionTracking.dataSetSpecificationId || "";
        this.batch = ingestionTracking.batch || "";
        this.objectName = ingestionTracking.objectName || "";
        this.batchReadyFolderPath = ingestionTracking.batchReadyFolderPath || "";

        this.createdBy = ingestionTracking.createdBy !== undefined ? ingestionTracking.createdBy : '';
        this.createdDate = ingestionTracking.createdDate;
        this.updatedBy = ingestionTracking.updatedBy !== undefined ? ingestionTracking.updatedBy : ''
        this.updatedDate = ingestionTracking.updatedDate;

        if (ingestionTracking.supplier !== undefined && ingestionTracking.supplier !== null) {
            this.supplier = new Supplier(ingestionTracking.supplier);
        }
    }
}
